### PR TITLE
Register `UiImageSize`

### PR DESCRIPTION
**Pull request by ickshonpe**
# Objective

Add `register_type` and derive `Reflect` for `UiImageSize`.

## Changelog
* Added `register_type` and derive `Reflect` for `UiImageSize`.

### CODE REVIEW

File: crates/bevy_ui/src/widget/image.rs [hunk 1]
```diff
@@ -8,14 +8,15 @@ use bevy_ecs::{
...
+use bevy_reflect::Reflect;
 use bevy_render::texture::Image;
 #[cfg(feature = "bevy_text")]
 use bevy_text::Text;
 
 /// The size of the image in pixels
 ///
 /// This field is set automatically
-#[derive(Component, Copy, Clone, Debug, Default)]
+#[derive(Component, Copy, Clone, Debug, Default, Reflect)]
```
Comment (MrGVSV) [inline crates/bevy_ui/src/widget/image.rs:19]: Nit: We should also derive `FromReflect` while we're here

Comment (MrGVSV) [inline crates/bevy_ui/src/widget/image.rs:19]: This should also add type data registrations like:
```
#[reflect(Component, FromReflect, Default)]
```

**Review by MrGVSV — CHANGES_REQUESTED** [review]

File: crates/bevy_ui/src/widget/image.rs [hunk 2]
```diff
@@ -5,17 +5,19 @@ use bevy_ecs::query::Without;
...
 use bevy_text::Text;
 
-/// The size of the image in pixels
+/// The size of the image in physical pixels
 ///
-/// This field is set automatically
-#[derive(Component, Copy, Clone, Debug, Default)]
+/// This field is set automatically by `update_image_calculated_size_system`
+#[derive(Component, Debug, Copy, Clone, Default, Reflect, FromReflect)]
+#[reflect(Component, Default)]
```
Comment (MrGVSV) [inline crates/bevy_ui/src/widget/image.rs:20]: Could we add `FromReflect` (`ReflectFromReflect`) to this list?

**Review by MrGVSV — CHANGES_REQUESTED** [review]

File: crates/bevy_ui/src/widget/image.rs (hunk 2 above)
Comment (ickshonpe) [reply crates/bevy_ui/src/widget/image.rs:20]: oh right I'm silly I need to learn how the reflection API works properly. I've mostly just been doing what I thought was needed to support reflection, without really understanding how it worked.

So what you mean is that this should derive `Reflect`, and then reflect `FromReflect`?

Most of the components in bevy_ui only derive `Reflect` and then reflect `Component` and `Default`, is this wrong?
I thought `FromReflect` was something that you needed to derive for types with a container field like an option or a vector.

**Review by MrGVSV — APPROVED** [review]
Looks good. Thanks!

**Review by Sheepyhead — APPROVED** [review]

**Review by mockersf — APPROVED** [review]

**Comment by mockersf** [comment]
could you rebase this PR? CI is failing for a rust update that has already been done on main

**Comment by ickshonpe** [comment]
Okay

**Pull request merged**